### PR TITLE
Add Rholang-Mercury tutorial doc

### PR DESCRIPTION
**Pull request by pyrocto**
This PR creates a `rholang/doc` directory and adds the tutorial.  The code in the tutorial uses the syntax in the 0.2 spec.  I'll be creating another version of the tutorial for the upcoming SDK release that uses the syntax supported by the current compiler.

### CODE REVIEW

**Comment by nashef** [comment]
Can you rename the file to rholangtut-mercury or something? Just so it's more obvious what we're looking at?

**Comment by pyrocto** [comment]
PTAL

File: rholang/doc/rholangtut-mercury.md [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+    3     for (@text <- chan) system!("print", text)
+    4 }
+
+1) To create a new, private channel, we use the `new ... in` construction. No other process can send or receive messages over this channel unless we explicitly send this channel to the other process.
+
+2) We send the string process `"Hello again, world!"` over the new channel.
+
+3) We listen on the new channel for a message.  The `for` operation blocks until there's a message available on the channel `chan`. 
+
+The language of patterns (also known as "spatial types") includes the language of processes and names.  Every process with free variables is a pattern that matches a process with the same structure; the free variables bind to the subprocesses in that position.  On line 3, we expect a tuple with one element.  That element is the serialization of some process, and therefore starts with an `@`.  Since we sent a string on line 2, the free variable`text` gets bound to that string process.  Finally, line 2 forwards that string to the `system` process to print.
```
Comment (tymm) [inline rholang/doc/rholangtut-mercury.md]: `Finally, line 2 forwards [...]`.
It's line 3.

**Comment by nashef** [comment]
LGTM

**Pull request merged**